debian/ubuntu-advantage-tools.postinst

```diff
@@ -4,14 +4,6 @@ set -e
 
 . /etc/os-release  # For VERSION_ID
```

Comment:
While we are dropping this content, I recognize that we have an expectation that both VERSION_ID and UBUNTU_CODENAME shall always come from /etc/os-release. Maybe we should be defensive about or expectations because we are exposed to two types of errors when UBUNTU_CODENAME is unset:
  1. unintended silent successes from `check_is_lts` due to it's use of grep -q "" 
  2. to an error in `check_is_active_esm` if invalid or empty series is specified
     `ubuntu-distro-info: invalid distribution series ''`

We have the same type of exposure to silent successes if VERSION_ID is unset thoughout the code. I'd prefer if we are explicit and loud in our errors with exit 1 from postinst rather than silently succeeding but not setting up ESM properly. The latter situation could result in an unpatched and insecure system as ESM.

I realize that exit 1 from our postinst is bad and breaks unattended-upgrades etc. So we'd have to weigh whether we want the bug here or not. 
```suggestion
. /etc/os-release  # For VERSION_ID and UBUNTU_CODENAME
if [ -z "${VERSION_ID}" -o -z "${UBUNTU_CODENAME}" ]; then
    echo "Error: missing either VERSION_ID or UBUNTU_CODENAME in /etc/os-release: VERSION_ID=${VERSION_ID}  UBUNTU_CODENAME=${UBUNTU_CODENAME}" 1>&2
    exit 1
elif
```

Reply:
It's fine to have no `VERSION_ID`, because it is used throughout the postinst to perform specific things to `xenial` - it is not even a silent fail, as Xenial will always define the `VERSION_ID` - it is success.
For 1. : This is true. I am changing the script to set the  `UBUNTU_CODENAME` to `UNKNOWN` if not found - all our LTSs should define it.
For 2. : This is not a problem if 1 is not a problem, because  `check_is_active_esm` is only executed if `check_is_lts` passes